### PR TITLE
Introduce ExternalSessionService.

### PR DESCRIPTION
**Pull request by demiankatz**
This migrates most of the external session handling logic to a new database service. The expiration cleanup command can be ported after #3714 is merged, since that work requires changes pending in that other PR.

All tests are passing, but I don't have a real-world environment to test this with, so help with that would be appreciated!

### CODE REVIEW

File: module/VuFind/src/VuFind/Db/Table/ExternalSession.php [hunk 1]
```diff
@@ -103,10 +107,12 @@ public function getByExternalSessionId($sid)
      * @param string $sid Session ID to erase
      *
      * @return void
+     *
+     * @deprecated Use ExternalSessionServiceInterface::addSessionMapping()
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Db/Table/ExternalSession.php]: This should point to `ExternalSessionServiceInterface::destroySession()`

File: module/VuFind/src/VuFind/Db/Table/ExternalSession.php [hunk 2]
```diff
@@ -91,10 +93,12 @@ public function addSessionMapping($localSessionId, $externalSessionId)
      * @param string $sid External session ID to retrieve
      *
      * @return ?\VuFind\Db\Row\ExternalSession
+     *
+     * @deprecated Use ExternalSessionServiceInterface::addSessionMapping()
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Db/Table/ExternalSession.php]: Wrong method name here.

File: module/VuFind/src/VuFind/Db/Service/ExternalSessionServiceInterface.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    ): ExternalSessionEntityInterface;
+
+    /**
+     * Retrieve an object from the database based on an external session ID
+     *
+     * @param string $sid External session ID to retrieve
+     *
+     * @return ?ExternalSessionEntityInterface
+     */
+    public function getExternalSessionById(string $sid): ?ExternalSessionEntityInterface;
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Db/Service/ExternalSessionServiceInterface.php]: I believe the old name was intentionally get**By**ExternalSessionId so as to not confuse it with the table id column.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

**Comment by EreMaijala** [comment]
@demiankatz I'll see if I can add a Mink test for this using SimulatedSSO.

**Comment by demiankatz** [comment]
Thanks for the review, @EreMaijala -- I've addressed your comments (including changing getExternalSessionById to getExternalSessionByExternalSessionId, which is long and redundant but more clear!).

Thanks also for the offer of test coverage -- more never hurts, though it does appear that we're at least using some parts of the service in ShibbolethLogoutNotificationTest.

**Comment by EreMaijala** [comment]
@demiankatz Indeed, but I think it's still worth adding it to SimulatedSSO test so that it's the "real thing" adding the session mapping.

**Comment by demiankatz** [comment]
I'm currently working on adding the expiration command...

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**